rfcs/45-stageless.md

```diff
@@ -0,0 +1,727 @@
+# Feature Name: `stageless`
+
+## Summary
+
+The existing stage boundaries result in a large number of complications as they prevent us from operating across stages.
+Moreover, several related features (run criteria, states and fixed timesteps) are rather fragile and overly complex.
+This tangled mess needs to be holistically refactored: transferring the ownership of systems to an undivided `Schedule`, simplifying run criteria, taking full advantage of system labels as sets and leveraging exclusive systems to handle complex logic.
+
+## Motivation
+
+The fundamental challenges with the current stage-driven scheduling abstraction are [numerous](https://github.com/bevyengine/bevy/discussions/2801).
+Of particular note:
+
+- Dependencies do not work across stages.
+- States do not work across stages.
+- Plugins can add new, standalone stages.
+- Run criteria (including states and fixed timestep run criteria!) cannot be composed.
+- Run criteria cannot be reused across states or stages.
+- The stack-driven state model is overly elaborate and does not enable enough use cases to warrant its complexity.
+- The architecture for turn-based games and other even slightly unusual patterns is not obvious.
+- We cannot easily implement a builder-style strategy for configuring systems.
+
+Unfortunately, all of these problems are deeply interwoven. Despite our best efforts, fixing them incrementally at either the design or implementation stage is impossible, as it results in myopic architecture choices and terrible technical debt.
+
+## User-facing explanation
+
+This explanation is, in effect, user-facing documentation for the new design.
+In addition to a few new concepts, it throws out much of the current system scheduling design that you may be familiar with.
+The following elements are substantially reworked:
+
+- schedules (simplified, each `World` can store multiple schedules in a `SystemRegistry` resource)
+- run criteria (can no longer loop, are now systems)
+- states (simplified, no longer purely run-criteria powered)
+- fixed time steps (no longer a run criterion)
+- exclusive systems (no longer special-cased)
+- command processing (now performed in a `flush_commands` exclusive system)
+- system labels and system sets (merged, can now be directly configured)
+- stages (yoten)

```

Comment:
Meta: "yoten" will mean nothing to a huge percentage of the worlds' population and may not stand the test of time. I'd prefer it if we used clear wording here.

Reply:
```suggestion
- stages (removed)
```